lib/puppet/provider/openldap_overlay/olc.rb

```diff
@@ -54,11 +66,31 @@ def create
     t << "changetype: add\n"
     t << "objectClass: olcConfig\n"
     t << "objectClass: olcOverlayConfig\n"
-    t << "objectClass: olcMemberOf\n" if resource[:overlay] == 'memberof'
+    case resource[:overlay]
+    when 'memberof'
+      t << "objectClass: olcMemberOf\n"
+    when 'ppolicy'
+      t << "objectClass: olcPPolicyConfig\n"
+    when 'dynlist'
+      t << "objectClass: olcDynamicList\n"
+    when 'auditlog'
+      t << "objectClass: olcAuditLogConfig\n"
+    when 'constraint'
+      t << "objectClass: olcConstraintConfig\n"
+    when 'pcache'
+      t << "objectClass: olcPcacheConfig\n"
+    end
+    #t << "objectClass: olcMemberOf\n" if resource[:overlay] == 'memberof'
     t << "olcOverlay: #{resource[:overlay]}\n"
+    if resource[:options]
+      resource[:options].each do |opt|
+        t << opt.split(':')[0] + ": " + opt.split(':', 2)[1]
+      end
+    end
     t.close
     Puppet.debug(IO.read t.path)
     begin
+      system "cat " + t.path
```

Comment:
if this is for debugging, please use puppet's `--debug` option, we already print the output with `Puppet.debug(IO.read t.path)`